File: solutions/typescript/VB_T3_006_option_unwrap_or.ts
// VB-T3-006: Option Unwrap Or -- TypeScript baseline
type Option = { tag: "None" } | { tag: "Some"; value: number };
function optionUnwrapOr(opt: Option, defaultVal: number): number {
  switch (opt.tag) { case "None": return defaultVal; case "Some": return opt.value; }
}
console.assert(optionUnwrapOr({ tag: "Some", value: 42 }, 0) === 42);
console.assert(optionUnwrapOr({ tag: "None" }, 99) === 99);

File: solutions/python/VB_T3_006_option_unwrap_or.py
"""VB-T3-006: Option Unwrap Or -- Python baseline."""
class Option: pass
class NoneOpt(Option): pass
class Some(Option):
    __match_args__ = ("value",)
    def __init__(self, value: int): self.value = value

def option_unwrap_or(opt: Option, default: int) -> int:
    match opt:
        case NoneOpt(): return default
        case Some(v): return v

if __name__ == "__main__":
    assert option_unwrap_or(Some(42), 0) == 42
    assert option_unwrap_or(NoneOpt(), 99) == 99
    print("VB-T3-006 ok")
